src/L2Controller.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.21;
+
+import { IERC20 }   from "forge-std/interfaces/IERC20.sol";
+import { IERC4626 } from "forge-std/interfaces/IERC4626.sol";
+
+import { AccessControl } from "openzeppelin-contracts/contracts/access/AccessControl.sol";
+
+import { IALMProxy } from "src/interfaces/IALMProxy.sol";
+
+interface IPSM3Like {
+    function asset0() external view returns(address);
+    function asset1() external view returns(address);
+    function asset2() external view returns(address);
+}
+
+interface ISNSTLike is IERC4626 {
+    function nst() external view returns(address);
+}
+
+contract L2Controller is AccessControl {
+
+    // TODO: Inherit and override interface
+
+    /**********************************************************************************************/
+    /*** State variables                                                                        ***/
+    /**********************************************************************************************/
+
+    bytes32 public constant FREEZER = keccak256("FREEZER");
+    bytes32 public constant RELAYER = keccak256("RELAYER");
+
+    IALMProxy public immutable proxy;
+    IPSM3Like public immutable psm;
+
+    IERC20    public immutable nst;
+    IERC20    public immutable usdc;
+    ISNSTLike public immutable snst;
```

Comment:
`snst` on all foreign domains is going to be ERC20.

Reply:
Updated